server/controllers/DocumentControllers.js

```diff
@@ -0,0 +1,215 @@
+  },
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns the list of documents available
+   */
+
+  listDocuments(req, res) {
+    const limit = req.query.limit || 10;
+    const offset = req.query.offset || 0;
+
+    if (isNaN(limit) || isNaN(offset)) {
+      return res.status(400).send({
+        message: 'limit and offset must be an integer'
+      });
+    }
+    if (req.query) {
+      return Document.findAndCount({
+        limit,
+        offset,
+      })
+    .then(document => res.status(200).send({
+      pagination: {
+        row: document.rows,
+        paginationDetails: pagination(document.count, limit, offset)
+      }
+    }))
+    .catch(() => res.status(403).send({
+      message: 'limits and offsets must be number'
+    }));
+    }
+  },
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns the requested document
+   */
+  getDocument(req, res) {
+    if (isNaN(req.params.id)) {
+      return res.status(400).send({
+        message: 'Id must be an integer'
+      });
+    }
+    return User
+    .findById(res.locals.decoded.UserId, {
+      include: [{
+        model: Document,
+        as: 'documents',
+      }],
+      attributes: ['fullname', 'username', 'email']
+    })
+    .then((user) => {
+      if (!user) {
+        return res.status(404).send({
+          message: 'User Not Found',
+        });
+      }
+      return res.status(200).send({
+        message: `These are the documents created by ${res.locals.decoded.userFullname}`,
+        user,
+      });
+    })
+    .catch(() => res.status(400).send({
+      message: 'Oops! something went wrong, Please try again'
+    }));
+  },
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns the updated document
+   */
+  updateDocument(req, res) {
+    if (!Number.isInteger(Number(req.params.id))) {
+      return res.status(400).json({
+        message: 'Invalid document id'
+      });
+    }
+    return Document
+      .find({
+        where: {
+          id: req.params.id,
+        },
+      })
+      .then((document) => {
+        if (!document) {
+          return res.status(404).send({
+            message: 'The Document does not exist',
+          });
+        }
+        if (req.body.title) {
+          req.body.title = (req.body.title).toLowerCase();
+        }
+        return document
+          .update(req.body, { fields: Object.keys(req.body) })
+          .then(() => res.status(200).send({
+            message: 'The Document successfully updated',
+            id: document.id,
+            title: document.title,
+            content: document.content,
+            access: document.access,
+            document,
+          }))
+          .catch(() => res.status(400).send({
+            message: 'Document not found',
+          }));
+      })
+      .catch(() => res.status(400).send({
+        message: 'Problem encountered, please try again',
```

Comment:
Can you be more specific about what problem it is that occurred? That way the developers consuming your API can be better informed.